components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/OAuth2JWTTokenValidator.java

```diff
@@ -181,7 +198,20 @@ private IdentityProvider getResidentIDPForIssuer(String jwtIssuer, String tenant
                     OIDC_IDP_ENTITY_ID).getValue();
         }
 
-        if (!jwtIssuer.equals(issuer)) {
+        String resourceAccessOrgId = getOrganizationManager().resolveOrganizationId(tenantDomain);
+        if (StringUtils.isNotEmpty(resourceResidentOrgId) && !resourceResidentOrgId.equals(resourceAccessOrgId)) {
+            // Check the tenant relationship if the token is not issued for the same tenant.
+            String issuerURLChecker = jwtIssuer.replace("/t/" + tenantDomain + "/",
+                    "/o/" + resourceResidentOrgId + "/");
+            List<String> resourceResidentOrgAncestors = getOrganizationManager().getAncestorOrganizationIds(
+                    resourceResidentOrgId);
+            // TODO: This logic can be improved after adding a config var to detect whether
```

Comment:
This utility function is available now, remove the todo and fix properly

Reply:
Resolved in https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2086/commits/e5b792101b4251e33c6454ad4ae069aafdfe2c12